pytorch_lightning/trainer/connectors/slurm_connector.py

```diff
@@ -1,61 +0,0 @@
-import logging
-import os
-import signal
-from subprocess import call
-
-log = logging.getLogger(__name__)
-
-
-class SLURMConnector:
```

Comment:
the signal handling is not tested, so no surprise there are no failing tests. 
But as you can see in the code below, this handles automatic job resubmission, and if you remove it, Lightning will lose support for this feature.

Reply:
is it expected that lightning needs to natively handle automatic job resubmission? or does it make sense for this to be part of the application using Lightning to handle?

Reply:
Yes, it is expected and a pretty important feature of the Lightning framework :)

Reply:
- would it make sense for this to be part of the SLURMEnvironment if users wanted to disable automatic resubmission? 
- should resubmission be opt-in instead of opt-out?

Reply:
> would it make sense for this to be part of the SLURMEnvironment if users wanted to disable automatic resubmission?

yes
#6303